src/vm/methodtablebuilder.cpp

```diff
@@ -6962,13 +6962,28 @@ MethodTableBuilder::NeedsNativeCodeSlot(bmtMDMethod * pMDMethod)
 
 #ifdef FEATURE_TIERED_COMPILATION
     // Keep in-sync with MethodDesc::DetermineAndSetIsEligibleForTieredCompilation()
-    if (g_pConfig->TieredCompilation() &&
+    if ((g_pConfig->TieredCompilation() &&
 
         // Policy - If QuickJit is disabled and the module does not have any pregenerated code, the method would be ineligible
         // for tiering currently to avoid some unnecessary overhead
         (g_pConfig->TieredCompilation_QuickJit() || GetModule()->HasNativeOrReadyToRunImage()) &&
 
-        (pMDMethod->GetMethodType() == METHOD_TYPE_NORMAL || pMDMethod->GetMethodType() == METHOD_TYPE_INSTANTIATED))
+        (pMDMethod->GetMethodType() == METHOD_TYPE_NORMAL || pMDMethod->GetMethodType() == METHOD_TYPE_INSTANTIATED)) 
+
+#ifdef FEATURE_REJIT
+        ||
+
+        // Methods that are R2R need precode if ReJIT is enabled. Keep this in sync with MethodDesc::IsEligibleForReJIT()
+        (ReJitManager::IsReJITEnabled() &&
+
+            (GetMethodClassification(pMDMethod->GetMethodType()) == mcIL
+                || GetMethodClassification(pMDMethod->GetMethodType()) == mcDynamic) &&
+
+            !pMDMethod->GetOwningType()->GetModule()->IsCollectible() &&
```

Comment:
What's the difference between `pMDMethod->GetOwningType()->GetModule()` and `GetModule()` (which is used above)? Just wondering if there is a difference and if the existing usage should also be changed, or if there is no difference then it would be nice to use a consistent method to get the module.

Reply:
I don't think there is a difference